src/index.cpp

```diff
@@ -2636,6 +2682,10 @@ template <typename T, typename TagT, typename LabelT> void Index<T, TagT, LabelT
         reposition_points((uint32_t)_max_points, (uint32_t)_nd, (uint32_t)_num_frozen_pts);
         _start = (uint32_t)_nd;
     }
+    if (_filtered_index && _dynamic_index)
```

Comment:
why should this be called AND line 2682 be called? wont it reposition twice?

Reply:
I see the confusion here, the function ```reposition_points``` on line 2682, actually reposition all points where as reposition_medoids just updates the id's of medoids as fz_points = medoids have been moved.

Reply:
the function has been removed and logic is now incorporated into reposition_points.